docs/utility.rst

```diff
@@ -0,0 +1,227 @@
+ >
+
+You can see here that the ``simple_image_1.lbl`` file can be
+loaded and the resulting Python object encoded with each of the
+PVL dialects that the ``pvl`` library knows.
+
+A file with broken PVL text::
+
+ > pvl_validate tests/data/pds3/broken/broken1.lbl
+ PDS3 | does NOT load |
+ ODL  | does NOT load |
+ PVL  | does NOT load |
+ ISIS |     Loads     |     Encodes
+ Omni |     Loads     |     Encodes
+ >
+
+Here, the PVL text in broken1.lbl cannot be loaded by the PDS3, ODL, or PVL dialects, to learn why
+use ``-v``::
+
+ > pvl_validate -v tests/data/pds3/broken/broken1.lbl
+ ERROR: PDS3 load error tests/data/pds3/broken/broken1.lbl (LexerError(...), 'Expecting an Aggregation Block, an Assignment Statement, or an End Statement, but found "=" : line 3 column 7 (char 23)')
+ ERROR: ODL load error tests/data/pds3/broken/broken1.lbl (LexerError(...), 'Expecting an Aggregation Block, an Assignment Statement, or an End Statement, but found "=" : line 3 column 7 (char 23)')
+ ERROR: PVL load error tests/data/pds3/broken/broken1.lbl (LexerError(...), 'Expecting an Aggregation Block, an Assignment Statement, or an End Statement, but found "=" : line 3 column 7 (char 23)')
+ PDS3 | does NOT load |
+ ODL  | does NOT load |
+ PVL  | does NOT load |
+ ISIS |     Loads     |     Encodes
+ Omni |     Loads     |     Encodes
+
+This tells us that in these cases, there is a parameter with a
+missing value.  However, the OmniParser (the default, and also what
+the ISIS dialect uses) has more tolerance for broken PVL text, and
+is able to load it, and then write valid PVL back out.
+
+Here's a file which has some PVL text which is valid for some dialects, but not others::
+
+ > pvl_validate tests/data/pds3/dates.lbl
+ PDS3 |     Loads     | does NOT encode
+ ODL  |     Loads     |     Encodes
+ PVL  |     Loads     |     Encodes
+ ISIS |     Loads     |     Encodes
+ Omni |     Loads     |     Encodes
+ >
+
+Here, ``pvl_validate`` indicates that it can load the file with all of the PVL dialects, and
+can encode it back for most.  What was the problem::
+
+ > pvl_validate -v tests/data/pds3/dates.lbl
+ ERROR: PDS3 encode error tests/data/pds3/dates.lbl PDS labels should only have UTC times, but this time has a timezone: 01:12:22+07:00
+ PDS3 |     Loads     | does NOT encode
+ ODL  |     Loads     |     Encodes
+ PVL  |     Loads     |     Encodes
+ ISIS |     Loads     |     Encodes
+ Omni |     Loads     |     Encodes
+
+
+It indicates that it cannot encode the Python object out to the
+PDS3 format because it contains a date with a different time zone
+(which aren't allowed in a PDS3 Label).  So this is an example of
+how the loaders are a little more permissive, but to really test
+whether some PVL text is conformant, it also should be able to be
+encoded.
+
+In this case, if the user wants to write out a valid PDS3 label, they will have to do 
+some work to fix the value.
+
+
+Validating a bunch of files::
+
+ > pvl_validate tests/data/pds3/*lbl
```

Comment:
wow this is a cool capability!

Reply:
Yeah, the benefit of having a more strict set of PVL definitions means that this is possible.